### PR TITLE
Add support for Codespaces

### PR DESCRIPTION
**Pull request by LukasReschke**
This allows setting up a Nextcloud instance within Visual Studio Code (locally) or GitHub code spaces (cloud) within a minute.

Just running the following, should result in a running Nextcloud instance: 

```
php -S 0.0.0.0:80
```

In addition, this ships with:

**XDebug**
Screenshot 2021-02-08 at 17 07 52

**Language Server**
Screenshot 2021-02-08 at 17 46 05

**MailHog**
Screenshot 2021-02-08 at 17 44 40

**Adminer**
Screenshot 2021-02-08 at 17 44 24

### CODE REVIEW

File: .devcontainer/setup.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )/../" >/dev/null 2>&1 && pwd )"
+
+cd $DIR/
+git submodule update --init
+apt-get update -y
+apt-get install --no-install-recommends -y \
+    php7.4 \
```
Comment (ChristophWurst) [inline .devcontainer/setup.sh:8]: why not 8?

**Review by ChristophWurst — APPROVED** [review]

File: .devcontainer/setup.sh (hunk 1 above)
Comment (LukasReschke) [reply .devcontainer/setup.sh:8]: Is there already a PHP 8.0 compatible open-source language server? https://github.com/felixfbecker/php-language-server/issues/798 at least seems to not support 8.0

**Review by GretaD — APPROVED** [review]

**Review by nickvergessen — APPROVED** [review]
🐘

File: .devcontainer/setup.sh (hunk 1 above)
Comment (nickvergessen) [reply .devcontainer/setup.sh:8]: php7.4-intl should be required nowadays (still listed "recommended" in https://docs.nextcloud.com/server/stable/admin_manual/installation/source_installation.html ) as it helps on emoji handling

Comment (go2sh) [reply .devcontainer/setup.sh:8]: Just for your reference: There is some support in https://github.com/bmewburn/vscode-intelephense

**Comment by kesselb** [comment]
Another pull request to add a devcontainer: https://github.com/nextcloud/server/pull/22973

**Comment by LukasReschke** [comment]
> Another pull request to add a devcontainer: #22973

Didn't see that one. What bothers me there is the amount of third-party extensions included, for me that would certainly be a no-go. (considering they can access your GitHub session)

**Comment by LukasReschke** [comment]
@rullzer Looks okay? If so let's merge that. This allows setting up a functional dev environment in a few clicks :)

**Pull request merged**